aws-ecs/terraform/variables.tf

```diff
@@ -0,0 +1,129 @@
+variable "vpc_id" {
+  description = "VPC ID for deployment"
+  type        = string
+}
+
+variable "public_subnets" {
+  description = "List of public subnets for deployment"
+  type        = list(string)
+  default     = []
+}
+
+variable "private_subnets" {
+  description = "List of private subnets for deployment"
+  type        = list(string)
+  default     = []
+}
+
+variable "default_vpc" {
+  description = "Set true if you want to use default VPC or VPC without private networks."
+  type        = bool
+  default     = false
+}
+
+variable "create_lb" {
+  description = "Set to true if you want to publish services publicly via Application Load Balancer"
+  type        = bool
+  default     = false
+}
+
+variable "existing_lb_arn" {
+  description = "Set ALB ARN if you want to use existing Application Load Balancer for chosen deployment."
+  type        = string
+  default     = null
+}
+
+variable "launch_type" {
+  description = "Launch type for ECS Services. Set to EC2 if you've provided cluster_name variable with existing EC2 cluster name."
+  type        = string
+  default     = "FARGATE"
+}
+
+variable "controller_settings" {
+  type = map(string)
```

Comment:
may be reasonable to use  `structural type` here https://www.terraform.io/language/expressions/type-constraints#structural-types